samtools: 1.10 -> 1.11 , lumpy: 0.3.0 -> 0.3.1

unode opened the pull request:
###### Motivation for this change

Needs #100812 to be merged first or the wrong version of `htslib` will be used.

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

unode committed:
samtools: 1.10 -> 1.11

SuperSandro2000 commented:
Fails to build for me on NixOS:

```
gcc -Wall -g -O2 -I. -I/nix/store/yykmsrs5gnzhcihlrpawbhwamyvhha13-htslib-1.10.2/include -I./lz4  -c -o sample.o sample.c
bam_fastq.c: In function 'copy_tag':
bam_fastq.c:267:15: warning: implicit declaration of function 'bam_aux_get_str'; did you mean 'bam_aux_get'? [-Wimplicit-function-declaration]
  267 |     int ret = bam_aux_get_str(rec, tag, linebuf);
      |               ^~~~~~~~~~~~~~~
      |               bam_aux_get
sam_view.c: In function 'main_samview':
sam_view.c:468:47: warning: implicit declaration of function 'fai_path'; did you mean 'fai_fetch'? [-Wimplicit-function-declaration]
  468 |     if (fn_fai == 0 && ga.reference) fn_fai = fai_path(ga.reference);
      |                                               ^~~~~~~~
      |                                               fai_fetch
sam_view.c:468:45: warning: assignment to 'char *' from 'int' makes pointer from integer without a cast [-Wint-conversion]
  468 |     if (fn_fai == 0 && ga.reference) fn_fai = fai_path(ga.reference);
      |                                             ^
sam_view.c:682:13: error: 'errno' undeclared (first use in this function)
  682 |             errno = 0;
      |             ^~~~~
sam_view.c:43:1: note: 'errno' is defined in header '<errno.h>'; did you forget to '#include <errno.h>'?
   42 | #include "bedidx.h"
  +++ |+#include <errno.h>
   43 |
sam_view.c:682:13: note: each undeclared identifier is reported only once for each function it appears in
  682 |             errno = 0;
      |             ^~~~~
bam_sort.c: In function 'seq_to_bam':
bam_sort.c:2163:9: warning: implicit declaration of function 'bam_set_seqi'; did you mean 'bam_get_seq'? [-Wimplicit-function-declaration]
 2163 |         bam_set_seqi(seq, i, seq_nt16_table[(unsigned char)buf[i]]);
      |         ^~~~~~~~~~~~
      |         bam_get_seq
make: *** [Makefile:135: sam_view.o] Error 1
make: *** Waiting for unfinished jobs....
builder for '/nix/store/9zppbpar10asqd4iv7fa4pj5vfb46s8l-samtools-1.11.drv' failed with exit code 2
error: build of '/nix/store/9zppbpar10asqd4iv7fa4pj5vfb46s8l-samtools-1.11.drv' failed
```

unode commented:
@SuperSandro2000 can you try this after #100812 ? I suspect that's the cause of this error.

SuperSandro2000 commented:
@unode ping me if it is merged

unode commented:
@SuperSandro2000 #100812 was just merged, whenever possible can you give this one another try? Thanks!

SuperSandro2000 commented:
Result of `nixpkgs-review pr 100814` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>13 packages failed to build:</summary>
  <ul>
    <li>deeptools</li>
    <li>lumpy</li>
    <li>python27Packages.HTSeq</li>
    <li>python27Packages.pysam</li>
    <li>python37Packages.HTSeq</li>
    <li>python37Packages.cnvkit</li>
    <li>python37Packages.pysam</li>
    <li>python38Packages.HTSeq</li>
    <li>python38Packages.cnvkit</li>
    <li>python38Packages.pysam</li>
    <li>samtools</li>
    <li>tebreak</li>
    <li>truvari</li>
  </ul>
</details>

unode commented:
Thanks! Will work on getting the other packages fixed.

SuperSandro2000 commented:
> Thanks! Will work on getting the other packages fixed.

I think you need to rebase this PR and to pull in the other PR.

SuperSandro2000 commented on pkgs/applications/science/biology/samtools/default.nix:
```diff
@@ -2,11 +2,11 @@
-  version = "1.10";
+  version = "1.11";
 
   src = fetchurl {
```
Can you use fetchGithub please?

unode replied:
Unfortunately not in this case. The automatic package generated by GitHub is incomplete as stated by the authors on the [releases page](https://github.com/samtools/samtools/releases).

SuperSandro2000 replied:
We are not using the bundled library. They generated files are used and would be pointless to regenerate, right?

unode committed:
lumpy: 0.3.0 -> 0.3.1

unode committed:
pythonPackages.pysam: Disable failing tests unrelated to packaging

These will be fixed upstream and are due to format changes introduced
in samtools and htslib 1.11

unode commented:
After rebasing I was still getting test failures from `pysam` which are related to #100823 . Since I maintain that package too, I included the fix here.
While checking why lumpy was failing I also ended up updating it here too.

unode commented:
nixpkgs-review now reports 13 successes on my end.
@SuperSandro2000 would you mind checking again? thanks!

jbedo reviewed (changes requested):
Lumpy compiles but does not function:
```/nix/store/a3zwdl2m606mynzxcqa77ymk6a3j9nnf-python-2.7.18/bin/python: can't open file '/nix/store/wpzji1i2wgnn9b9y6jkqwpclp15w5bvf-lumpy-0.3.1//scripts/bamkit/bamlibs.py': [Errno 2] No such file or directory```

Error is caused by the new bamkit submodule, so fix is to fetch submodules as well.

Samtools tested fine for me.

unode commented:
Had to include an additional `sed` for htslib includes. @jbedo can you check on your end?

SuperSandro2000 commented:
Result of `nixpkgs-review pr 100814` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>10 packages failed to build:</summary>
  <ul>
    <li>deeptools</li>
    <li>lumpy</li>
    <li>python37Packages.HTSeq</li>
    <li>python37Packages.cnvkit</li>
    <li>python37Packages.pysam</li>
    <li>python38Packages.HTSeq</li>
    <li>python38Packages.cnvkit</li>
    <li>python38Packages.pysam</li>
    <li>tebreak</li>
    <li>truvari</li>
  </ul>
</details>
<details>
  <summary>3 packages built:</summary>
  <ul>
    <li>python27Packages.HTSeq</li>
    <li>python27Packages.pysam</li>
    <li>samtools</li>
  </ul>
</details>

SuperSandro2000 commented:
Result of `nixpkgs-review pr 100814` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>9 packages failed to build:</summary>
  <ul>
    <li>deeptools</li>
    <li>python37Packages.HTSeq</li>
    <li>python37Packages.cnvkit</li>
    <li>python37Packages.pysam</li>
    <li>python38Packages.HTSeq</li>
    <li>python38Packages.cnvkit</li>
    <li>python38Packages.pysam</li>
    <li>tebreak</li>
    <li>truvari</li>
  </ul>
</details>
<details>
  <summary>4 packages built:</summary>
  <ul>
    <li>lumpy</li>
    <li>python27Packages.HTSeq</li>
    <li>python27Packages.pysam</li>
    <li>samtools</li>
  </ul>
</details>

jbedo commented:
> Had to include an additional `sed` for htslib includes. @jbedo can you check on your end?

Lumpy works perfectly now, thanks!

jbedo reviewed (approved):
LGTM. FWIW `nix-review` is building all packages successfully for me on x86_64-linux.

SuperSandro2000 commented:
Result of `nixpkgs-review pr 100814` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>8 packages marked as broken and skipped:</summary>
  <ul>
    <li>deeptools</li>
    <li>python27Packages.HTSeq</li>
    <li>python37Packages.HTSeq</li>
    <li>python37Packages.cnvkit</li>
    <li>python38Packages.HTSeq</li>
    <li>python38Packages.cnvkit</li>
    <li>tebreak</li>
    <li>truvari</li>
  </ul>
</details>
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>samtools</li>
  </ul>
</details>

unode commented:
Hum... I don't have a darwin system to test on. @SuperSandro2000 can you pull any logs out?

SuperSandro2000 commented:
@unode those packages have a ``meta.broken = true`` for darwin so they are not build. I can try to build them if you want but I would move this to a separate PR.

unode commented:
@SuperSandro2000 ah thanks! Give them a try in case the updates fixed some of them but I agree we should move to a different PR.

Then any objections or do we have green light here? If so, I'll ask for a merge.

SuperSandro2000 commented:
I am not sure why the linux nixpkgs-review is so flaky. I build all packages manually and they succeeded. Maybe something about machine is not quite right.

unode commented:
> I am not sure why the linux nixpkgs-review is so flaky. I build all packages manually and they succeeded. Maybe something about machine is not quite right.

I had similar issues here. Still getting used to it. In a few cases had to modify the code to trigger a rebuild. It might just be the way it works with the nixpkgs cache...

SuperSandro2000 commented:
I just build the packages on darwin:

python27Packages.pysam
python38Packages.pysam
lumpy
deeptools
python27Packages.HTSeq
python38Packages.HTSeq
python38Packages.cnvkit

But please mark this in a separate PR.
